### ISSUE TITLE
if there is some cat operation in the model, libtorch1.4.0 can only deploy the model in cpu but not in gpu

### ISSUE THREAD

**Issue by mmchen88** [pytorch/pytorch#35462]
## 🐛 Bug

if there were some cat operations in the model, libtorch1.4.0 can only deploy the model in cpu, but not in gpu.

## To Reproduce

1.use pytorch1.4.0 to trace a model which has cat operations, make sure the model is GPU mode
2.use libtorch1.4.0 to deploy the traced model, it isnot ok( when the program goes here:
auto CAB_output = CAB_model.forward(inputs).toTensor();
Visual Studio outputs a dialog which says:
0x00007FF839BCA839 处有未经处理的异常: Microsoft C++ 异常: std::runtime_error，位于内存位置 0x00000070A1326658 处。)
3.make the pytorch to cpu mode, and trace it again ,then use libtorch1.4.0 to deploy the traced model, it is ok

i tried there models which have cat operations, here is one simple example:

class CAB(nn.Module):
    def __init__(self, in_channels, out_channels):
        super(CAB, self).__init__()
        self.global_pooling = nn.AdaptiveAvgPool2d(output_size=1)
        self.conv1 = nn.Conv2d(in_channels, out_channels, kernel_size=1, stride=1, padding=0)
        self.relu = nn.ReLU()
        self.conv2 = nn.Conv2d(out_channels, out_channels, kernel_size=1, stride=1, padding=0)
        self.sigmod = nn.Sigmoid()

    def forward(self, x1, x2):
        #x1, x2 = x  # high, low
        x = torch.cat([x1, x2], dim=1)
        x = self.global_pooling(x)
        x = self.conv1(x)
        x = self.relu(x)
        x = self.conv2(x)
        x = self.sigmod(x)
        x2 = x * x2
        res = x2 + x1
        return res

## Expected behavior

this model could be deployed in gpu mode by libtorch1.4.0
## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.4.0
 - OS (e.g., Linux): windows
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version:1.4.0
 - CUDA/cuDNN version:10.1
 - GPU models and configuration:
 - Any other relevant information:

**Comment by peterjc123** [comment]
What do you mean by "not ok"? Could you please elaborate a little bit?

**Comment by peterjc123** [comment]
Thanks for the update. One more question, have you copied all the DLLs from libtorch to the location of the executable?

**Comment by mmchen88** [comment]
> What do you mean by "not ok"? Could you please elaborate a little bit? Which version (CPU/GPU) of libtorch are you using? Could you please share more details about how did you deploy your model through libtorch?

when the program goes here:
auto CAB_output = CAB_model.forward(inputs).toTensor();
Visual Studio outputs a dialog which says:
0x00007FF839BCA839 处有未经处理的异常: Microsoft C++ 异常: std::runtime_error，位于内存位置 0x00000070A1326658 处

my libtorch version is 1.4.0.
my total deploy code is here:

#define CPU_mode false

//CAB_main
int main()
{
#if CPU_mode
	std::string CAB_model_path = "./traced/traced_CAB_cpu.pt";
#else
	std::string CAB_model_path = "./traced/traced_CAB_cuda.pt";
#endif

	torch::jit::script::Module CAB_model;
	try {
		// Deserialize the ScriptModule from a file using torch::jit::load().
		CAB_model = torch::jit::load(CAB_model_path);
	}
	catch (const c10::Error& e) {
		std::cerr << "error loading the CAB model\n";
		return -1;
	}

	at::Tensor X1_tensor = torch::zeros({ 1, 64, 48, 48 });
	at::Tensor X2_tensor = torch::zeros({ 1, 64, 48, 48 });
	std::vector<torch::jit::IValue> inputs;
#if CPU_mode
	inputs.push_back(X1_tensor);
	inputs.push_back(X2_tensor);

#else
	auto X1_var = torch::autograd::make_variable(X1_tensor, false);
	auto X2_var = torch::autograd::make_variable(X2_tensor, false);
	inputs.push_back(X1_var.to(at::kCUDA));
	inputs.push_back(X2_var.to(at::kCUDA));
#endif
	auto CAB_output = CAB_model.forward(inputs).toTensor();
	c10::IntList size;
	size = CAB_output.sizes();
	std::cout << size.at(0) << "," << size.at(1) << "," << size.at(2) << "," << size.at(3) << std::endl;
	return 0;
}

**Comment by mmchen88** [comment]
> Thanks for the update. One more question, have you copied all the DLLs from libtorch to the location of the executable?

No, I just copied c10.dll, c10_cuda.dll, torch.dll, libiomp5md.dll nvToolsExt64_1.dll.
thank you ,just now i copied all the DLLs from libtorch, it works
